py/miniconf-mqtt/miniconf/discover.py

```diff
@@ -30,35 +33,40 @@ async def discover(
     Returns:
         A list of discovered client prefixes that match the provided filter.
     """
-    suffix = "/alive"
-
-    sub = {}
+    if isinstance(client, str):
+        async with Client(client, protocol=MQTTv5) as mqtt_client:
+            return await do_discovery(mqtt_client, prefix, rel_timeout, abs_timeout)
+    else:
+        return await do_discovery(client, prefix, rel_timeout, abs_timeout)
 
-    def on_subscribe(_client, mid, _qos, _props):
-        sub[mid].set_result(True)
 
+async def do_discovery(
+        client: Client, prefix: str, rel_timeout: float, abs_timeout: float
+    ) -> List[str]:
+    """ Do the discovery operation. Refer to `discover` doc strings for parameters. """
```

Comment:
```suggestion
    """Do the discovery operation. Refer to `discover` doc strings for parameters. """
```

Reply:
Fixed in 51d73266a37342b6646bcd9aafaab9c4b9edb983